addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -4262,6 +4268,31 @@ QUnit.test('mark channel as seen if last message is visible when switching chann
     );
 });
 
+QUnit.test('add custom filter should filter messages accordingly to selected filter', async function (assert) {
+    assert.expect(4);
+
+    this.data['mail.channel'].records.push({
+        id: 20,
+        name: "General"
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_fetch') {
+                const domainsAsStr = args.kwargs.domain.map(domain => domain.join(''));
+                assert.step(`message_fetch_${domainsAsStr.join('')}`);
+            }
+            return this._super(...arguments);
+        },
+    });
+    assert.verifySteps(['message_fetch_needaction=true'], "A message_fetch request should have been done for needaction messages");
+
+    // Open filter menu of control panel and select a custom filter (id = 0)
```

Comment:
what is id=0? where does it come from

Reply:
It's the only custom filter available

Reply:
Any chance we could we add a custom filter in test and use that one instead?

Reply:
Ideally should test result of filter (= matching message in message list) instead of mock rpc.